cmake/cplusplus.cpp

```diff
@@ -1,4 +1,13 @@
-#if __cplusplus < OPENDDS_TEST_CPLUSPLUS
+// msvc doesn't define __cplusplus by default for some reason
```

Comment:
https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170

Reply:
So the issue we saw earlier in standup where ace6tao3 was defaulting to C++03 was caused by this trying to pass `/Zc:__cplusplus`. When that option is manually set in visual studio, it works as intended. Trying to pass it via `CMAKE_CXX_FLAGS` to `try_compile` like the PR is currently trying to do results in the option being ignored because `try_compile` is overriding it. It seems like the correct way to either set the option using the `CXXFLAGS` environment variable or to make a `CMakeLists.txt` just for this cpp file. Neither are appealing to me. This combined with the fact that `/Zc:__cplusplus` doesn't work with all of the VS versions we claim to support, suggests this should  use `_MSVC_LANG`.